src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const groupKey = item[key];
+
+    if (!acc[groupKey]) {
+      acc[groupKey] = [];
+    }
+
+    acc[groupKey].push(item);
+
+    return acc;
+  }, {} as any);
```

Comment:
```suggestion
  }, {});
```
Try to apply types and avoid using `any`. Feel free to ask for help in the chat